### PR TITLE
feat(e2e): Adding base classes for writing E2E tests in IotHub

### PR DESCRIPTION
**Pull request by vinagesh**

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+            string testSettingsUserPath = Path.Combine(workingDirectory, "config", $"{userName}.config.json");
+            if (File.Exists(testSettingsUserPath))
+            {
+                testSettingsConfigBuilder.AddJsonFile(testSettingsUserPath);
+            }
+
+
+            // TODO: Find out why the Microsoft.Extensions.Configuration.Binder package does not have a version specified in the Packages.Data.props
+            // IConfiguration config = testSettingsConfigBuilder.Build();
```
Comment (barustum) [inline sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:52]: Are these commented code lines still needed?

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+
+namespace Azure.Iot.Hub.Service.Tests
+{
+    public class TestSettings
+    {
+        public const string IotHubEnvironmentVariablesPrefix = "IOTHUB";
```
Comment (azabbasi) [inline sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:15]: This should match the service directory name, so just "IOT"

**Review by barustum — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs (hunk 1 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:52]: They are needed but I could not get that package to work. Once I figure that out, I will un-comment those lines.

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs (hunk 2 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:15]: I see!! Will change. Thanks.

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            testSettingsConfigBuilder.AddJsonFile(testSettingsCommonPath);
+
+            string testSettingsUserPath = Path.Combine(workingDirectory, "config", $"{userName}.config.json");
+            if (File.Exists(testSettingsUserPath))
+            {
+                testSettingsConfigBuilder.AddJsonFile(testSettingsUserPath);
+            }
+
+
+            // TODO: Find out why the Microsoft.Extensions.Configuration.Binder package does not have a version specified in the Packages.Data.props
```
Comment (azabbasi) [inline sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:51]: We have given it a version in the private repo for ADT, you can take a look at that one.

**Review by azabbasi — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs (hunk 3 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:51]: Cool. Thanks.

**Comment by azabbasi** [comment]
can you also address the CI build failures?

File: sdk/iot/Azure.Iot.Hub.Service/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+launchSettings.json
```
Comment (drwill-ms) [inline sdk/iot/Azure.Iot.Hub.Service/.gitignore:1]: launchSettings.json is in the root .gitignore file in the public mono repo. This is another sign we need a rebase on master for this feature branch.

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs (hunk 3 above)
Comment (drwill-ms) [inline sdk/iot/Azure.Iot.Hub.Service/tests/TestSettings.cs:51]: Because you need a rebase on master.

**Review by drwill-ms — CHANGES_REQUESTED** [review]
Rebase feature branch on master first.

**Pull request closed without merging**